src/Integration/QualityProfiles/SonarQubeQualityProfileProvider.cs

```diff
@@ -0,0 +1,152 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.Integration.Persistence;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarQube.Client.Messages;
+using SonarQube.Client.Services;
+
+namespace SonarLint.VisualStudio.Integration.RuleSets
+{
+    internal class SonarQubeQualityProfileProvider : IQualityProfileProvider
+    {
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly ILogger logger;
+
+        public SonarQubeQualityProfileProvider(ISonarQubeService sonarQubeService, ILogger logger)
+        {
+            if (sonarQubeService == null)
+            {
+                throw new ArgumentNullException(nameof(sonarQubeService));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.sonarQubeService = sonarQubeService;
+            this.logger = logger;
+        }
+
+        public QualityProfile GetQualityProfile(BoundSonarQubeProject project, Language language)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            if (language == null)
+            {
+                throw new ArgumentNullException(nameof(language));
+            }
+
+            if (language != Language.CSharp && language != Language.VBNET)
+            {
+                throw new ArgumentOutOfRangeException(nameof(language));
+            }
+
+            var roslynProfileExporter = GetSonarQubeQualityProfile(project, language).GetAwaiter().GetResult();
```

Comment:
When we add support for other languages in the future we'll have to use the standard APIs to fetch the QP directly. I think the _RoslynProfileExporter_ has some special behaviour related to third-party Roslyn analysers, so we'd have to decide at that point whether to support it or not. Still, it's a problem for another day.

Reply:
RoslynProfileExporter  is only for C# and VB.Net. Tomorrow the support of JS/C++ and so on will require a different strategy (but I don't know which yet).